File: typescript/playwright/tests/api.test.ts
import { test, expect } from '@playwright/test';
import { BASE_URL } from './helpers/api.helpers';

test.describe('Health Check', () => {
    test('should return healthy status', async ({ request }) => {
        const response = await request.get(`${BASE_URL}/health`);
        expect(response.status()).toBe(200);
    });
});

File: java/restassured/src/test/java/com/boilerplate/tests/ApiTest.java
package com.boilerplate.tests;

import com.boilerplate.tests.helpers.ApiHelpers;
import io.restassured.RestAssured;
import org.junit.jupiter.api.Test;

import static org.hamcrest.Matchers.equalTo;

class ApiTest {

    @Test
    void healthCheck() {
        RestAssured.given()
                .baseUri(ApiHelpers.BASE_URL)
                .when()
                .get("/health")
                .then()
                .statusCode(200)
                .body("status", equalTo("ok"));
    }
}
